### ISSUE TITLE
FormatException: Invalid group(?<=\d)(?=(\d{3})+(?!\d))

### ISSUE THREAD

**Issue by bandit197950** [dart-lang/sdk#37321]
On Dart VM have `FormatException: Invalid group(?<=\d)(?=(\d{3})+(?!\d))`
on code: `final _triadesRegExp = RegExp(r'(?<=\d)(?=(\d{3})+(?!\d))');`

**Comment by lrhn** [comment]
Teagan, do you know if this is expected behavor?

**Comment by sstrickl** [comment]
It's not expected behavior, and it works when I try it out. Here's my quick test file:

```
void main() {
  final _triadesRegExp = RegExp(r'(?<=\d)(?=(\d{3})+(?!\d))');
  var matches = _triadesRegExp.allMatches("1234i2345");
  for (var match in matches) {
    print("${match.start},${match.end}: \"${match.group(1)}\"");
  }
}
```

and when running that file, I get the following output:

```
1,1: "234"
6,6: "345"
```

which matches my expectations.

Is it possible this is an older version of Dart that doesn't include lookbehind support? IIRC, that was first included in the 2.3.0 release. In any case, more context of the Dart environment being used would be useful if it is post-2.3.0 so I could check on that version.

**Comment by sstrickl** [comment]
For now, closing since this doesn't seem to be an issue in versions of Dart that support lookbehind.
